sdk/core/azure-core/src/main/java/com/azure/core/util/QueryParameter.java

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util;
+
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Represents a single parameter to be added to a query string.
+ *<p>
+ * If multiple values are added to a query string with the same name (case-insensitive), then the values will be
+ * appended at the end of the same {@link QueryParameter} with commas separating them.
+ */
+public class QueryParameter {
+    private final String name;
+
+    // this is the actual internal representation of all values
+    private final List<String> values;
+
+    // but we also cache it to faster serve our public API
+    private String cachedStringValue;
```

Comment:
Given this is a cached value, it may be worth it to make it either volatile or be an AtomicReference to prevent potential race conditions between nulling and retrieving it

Reply:
Made volatile. Observation: same code is in Header class.